pkg/handler/v1/provider/openstack.go

```diff
@@ -28,34 +28,44 @@ import (
 	"k8c.io/kubermatic/v2/pkg/provider"
 )
 
+type credential struct {
+	credential, username, password, tenant, tenantID, domain string
+}
+
+func auth(ctx context.Context, req OpenstackReq, userInfoGetter provider.UserInfoGetter, presetsProvider provider.PresetProvider) (*provider.UserInfo, *credential, error) {
```

Comment:
I'm not very convinced by the name of this function, can we find one that better represents what it does?

Reply:
can you suggest any?

Reply:
personally, I would not have grouped the call to `userInfoGetter` and `getCretendials` in the first instance :-), but for this particular function I would probably use something like `getAuthInfo`

Reply:
umm, I grouped it coz `userInfo` is only used to `getCretendials`.
renaming  `auth to `getAuthInfo`

Reply:
that's legitimate ;-)